File: Backend/Events/Converters/Parties/Digimons/InBattleConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Parties.Digimons;

namespace Backend.Events.Converters.Parties.Digimons;

public static class InBattleConverter
{
    public static InBattleDTO ToDTO(InBattle digimonInBattle) => new()
    {
        Condition = digimonInBattle.Condition,
        Strength = digimonInBattle.Strength,
        Defense = digimonInBattle.Defense,
        Speed = digimonInBattle.Speed,
        HP = VitalConverter.ToDTO(digimonInBattle.HP),
        MP = VitalConverter.ToDTO(digimonInBattle.MP)
    };
}

File: Frontend/src/events/converters/parties/digimons/in-battle.converter.ts
import type { InBattleDTO } from "@/events/dto/parties/digimons/in-battle.dto";
import type { InBattle } from "@/models/party/digimon/in-battle";
import { VitalConverter } from "./vital.converter";

export class InBattleConverter {
  public static convert(newInBattleDto: InBattleDTO | null): InBattle {
    return {
      condition: newInBattleDto?.condition ?? 0,
      strength: newInBattleDto?.strength ?? 0,
      defense: newInBattleDto?.defense ?? 0,
      speed: newInBattleDto?.speed ?? 0,
      hp: VitalConverter.convert(newInBattleDto?.hp ?? null),
      mp: VitalConverter.convert(newInBattleDto?.mp ?? null),
    };
  }
}
